hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieCleanConfig.java

```diff
@@ -179,6 +179,12 @@ public class HoodieCleanConfig extends HoodieConfig {
           + " table receives updates/deletes. Another reason to turn this on, would be to ensure data residing in bootstrap "
           + " base files are also physically deleted, to comply with data privacy enforcement processes.");
 
+  public static final ConfigProperty<Boolean> CLEANER_IGNORE_APPEND_WRITE_COMMITS = ConfigProperty
+      .key("hoodie.cleaner.ignore.append.write.commits")
+      .defaultValue(false)
+      .markAdvanced()
+      .withDocumentation("When set to true, cleaner will ignore partition affected by commits/delta commits. This is usefule for append write mode");
+
```

Comment:
Just disable the cleaning serivce for append mode?

Reply:
> Just disable the cleaning serivce for append mode?

We still need the clean for replace commits

Reply:
The clean can be put right after the clustering commit I think.

Reply:
Do you mean we only trigger clean after every successful replace commit（ including clustering and delete partition)?  In this way we may need to disable auto commit and async clean, then add extra steps after committing the replace commit

Reply:
Yeah, but we have no good manner to infer that config if the table service operations are all async. How about just config to disable the cleaning manually in the ingestion job?

Reply:
> Yeah, but we have no good manner to infer that config if the table service operations are all async. How about just config to disable the cleaning manually in the ingestion job?

That's why we introduce the above config `hoodie.cleaner.ignore.append.write.commits` in our inner version.  We don't need to care about whether the table services are sync/async, running in the streaming ingestions job or just batch job like delete partitions by sparksql. User just add a single config where the clean service is running. What do you think?

Reply:
I kind of think the option is verbose. Disable the cleaning service has the same function right?

Reply:
> I kind of think the option is verbose. Disable the cleaning service has the same function right?

True if there is no replace commit. However clustering is commonly used in append-only ingestion.

Reply:
But the offline clustering will take over the cleaning task, both Spark and Flink offline clustering and compaction would trigger cleaning now, Spark supports this only in recent master code.